ddl/ddl_api.go

```diff
@@ -1745,3 +1750,24 @@ func validateCommentLength(vars *variable.SessionVars, comment string, maxLen in
 	}
 	return comment, nil
 }
+
+func checkModifyingDefinition(ctx sessionctx.Context, ident ast.Ident, originalColName model.CIStr) error {
+	sql := fmt.Sprintf(`select * from %s.%s where %s is null; `, ident.Schema.L, ident.Name.L, originalColName.L)
+	rows, _, err := ctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQL(ctx, sql)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	recordCount := fmt.Sprintf(`select * from %s.%s where %s is not null; `, ident.Schema.L, ident.Name.L, originalColName.L)
+	recordRows, _, err := ctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQL(ctx, recordCount)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	if len(rows) != 0 {
+		ctx.GetSessionVars().StmtCtx.AppendWarning(errInvalidUseOfNull)
+		return errInvalidUseOfNull
+	}
+	log.Infof("Records: %d  Duplicates: %d  Warnings: %d", len(recordRows), len(rows), ctx.GetSessionVars().StmtCtx.WarningCount())
```

Comment:
debug message? can be removed

Reply:
@XuHuaiyu It will be canceled, and this message needs to be displayed on the mysql client.

Reply:
@ciscoxll What do you mean by saying "message needs to be displayed on the mysql client"? Per my understanding, `log.Infof` only writes log into file which you passed in when you start tide-server.